### ISSUE TITLE
observations Buffer float32 initialization

### ISSUE THREAD

**Issue by XMaster96** [DLR-RM/stable-baselines3#40]
I have taken a look at SB3 and it looks good so far.

But one thing I don’t like is that the Replay/Rollout observations Buffer is always initialized with `float32`. It uses a lot of memory when using a image observations space, what is not necessary because the image observations a in `uint8`. My proposal is to initialize the observations buffer with what ever `dtype` the observations have. This is quite easy to do because `spaces.Space`, already has the `dtype` information stored in it.

NOTE:
This issue is a bit related to https://github.com/DLR-RM/stable-baselines3/issues/37

**Comment by PartiallyTyped** [comment]
This is actually resolved with #28.

https://github.com/DLR-RM/stable-baselines3/pull/28/files#diff-6bab9d371810aab2ba78872c37946a77L161-L165

**Comment by PartiallyTyped** [comment]
This is actually resolved with #28.

```diff python
-        self.observations = np.zeros((self.buffer_size, self.n_envs,) + self.obs_shape, dtype=np.float32)
-        self.actions = np.zeros((self.buffer_size, self.n_envs, self.action_dim), dtype=np.float32)
-        self.next_observations = np.zeros((self.buffer_size, self.n_envs,) + self.obs_shape, dtype=np.float32)
+        # Check that the replay buffer can fit into the memory
+        if psutil is not None:
+            mem_available = psutil.virtual_memory().available
+        self.observations = np.zeros((self.buffer_size, self.n_envs,) + self.obs_shape, dtype=observation_space.dtype)
+        self.actions = np.zeros((self.buffer_size, self.n_envs, self.action_dim), dtype=action_space.dtype)
+        self.next_observations = np.zeros((self.buffer_size, self.n_envs,) + self.obs_shape,
+                                          dtype=observation_space.dtype)
        self.rewards = np.zeros((self.buffer_size, self.n_envs), dtype=np.float32)
        self.dones = np.zeros((self.buffer_size, self.n_envs), dtype=np.float32)
```

Here:
https://github.com/DLR-RM/stable-baselines3/pull/28/files#diff-6bab9d371810aab2ba78872c37946a77L161-L165

**Comment by XMaster96** [comment]
I wasn't aware of that, 
Thanks
